### PR TITLE
Add envFromSecrets support to sftpgo chart

### PR DESCRIPTION
**Pull request by ggramal**
Good day @sagikazarmark.

Made this small PR for sftpgo chart to support setting env variables from specific secret keys.

### CODE REVIEW

**Comment by sagikazarmark** [comment]
@ggramal is there a reason why `envFrom` doesn't work for you?

**Comment by ggramal** [comment]
@sagikazarmark the reason why envFrom doesnt work for me is that we have a tool that creates k8s secrets for db credentials. Those secrets have predefined key structure and they are consumed by all of our services. So in order not to jump through the hoops it would be nice to do something like this
```
envFromSecrets:
  - name: SFTPGO_DATA_PROVIDER__PASSWORD
    valueFrom:
      secretKeyRef:
        key: passwd
        name: postgres-owner-sftpgo
```
I also generally think clients of this chart should also have a more flexible way of setting envs from secrets

File: charts/sftpgo/templates/deployment.yaml [hunk 1]
```diff
@@ -38,6 +38,13 @@ spec:
             - sftpgo
             - serve
           env:
+            {{- range .Values.envFromSecrets }}
+            - name: {{ .name | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .valueFrom.secretKeyRef.name | quote }}
+                  key: {{ .valueFrom.secretKeyRef.key | quote }}
+            {{- end }}
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/deployment.yaml:47]: Let's move this to the very end of the `env` section so it comes as last in the order of precedence. Also, use `toYaml` and `with` instead to make it more generic (support configmaps, etc).

File: charts/sftpgo/values.yaml [hunk 2]
```diff
@@ -56,6 +56,15 @@ envFrom: []
 # See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
 env: {}
 
+# -- Additional environment variables values of which are populated from specific [secrets and keys](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)
+envFromSecrets: []
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:60]: Let's call this `envVars` instead.

File: charts/sftpgo/values.yaml [hunk 3]
```diff
@@ -56,6 +56,15 @@ envFrom: []
 # See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
 env: {}
 
+# -- Additional environment variables values of which are populated from specific [secrets and keys](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:59]: Let's link to [this](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) page instead like this:

```
# See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
```

I'd also add a comment to `env` saying that it's a simplified syntax to allow key-value pairs and that `envVars` allows a more detailed syntax.

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
I see. I'm not very happy with the name `envFromSecrets` TBH. It's really what `env` should be, but it's already taken. Let's call it `envVars` instead.

File: charts/sftpgo/templates/deployment.yaml (hunk 1 above)
Comment (ggramal) [reply charts/sftpgo/templates/deployment.yaml:47]: Ok good point on making it more generic @sagikazarmark , will do that. But i would still suggest to leave it at the top of `env` section to allow clients to use envs set from secrets or configmaps in key value envs. Something like this

```
env:
  SFTPGO_DATA_PROVIDER__CONNECTION_STRING: "postgres://$(DB_USER):$(DB_PASSWORD)@db-sftpgo:5432/sftpgo"

envVars:
  - name: DB_PASSWORD
    valueFrom:
      secretKeyRef:
        key: passwd
        name: postgres-owner-sftpgo
  - name: DB_USER
    valueFrom:
      secretKeyRef:
        key: username
        name: postgres-owner-sftpgo
```

File: charts/sftpgo/values.yaml (hunk 2 above)
Comment (ggramal) [reply charts/sftpgo/values.yaml]: @sagikazarmark  made the needed changes please review

File: charts/sftpgo/templates/deployment.yaml (hunk 1 above)
Comment (ggramal) [reply charts/sftpgo/templates/deployment.yaml]: made the needed changes please review

File: charts/sftpgo/values.yaml (hunk 3 above)
Comment (ggramal) [reply charts/sftpgo/values.yaml]: done

File: charts/sftpgo/templates/deployment.yaml (hunk 1 above)
Comment (sagikazarmark) [reply charts/sftpgo/templates/deployment.yaml]: I see. Well, even then, I'd move the envVars section to after the manually defined env vars as those shouldn't be overwritten.

File: charts/sftpgo/values.yaml [hunk 4]
```diff
@@ -52,9 +52,25 @@ volumeMounts: []
 # See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
 envFrom: []
 
-# -- Additional environment variables passed directly to containers.
-# See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
+# -- Allows to set additional environment variables. See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:55]:
```suggestion
# -- Allows setting additional environment variables.
# See the [API reference](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables) for details.
```

File: charts/sftpgo/values.yaml [hunk 5]
```diff
@@ -52,9 +52,25 @@ volumeMounts: []
...
+  #       name: SOME_SECRET_NAME
+  #       key: SOME_SECRET_KEY
+  # - name: SOME_ENV_VAR3
+  #   valueFrom:
+  #     configMapKeyRef:
+  #       name: SOME_CONFIGMAP_NAME
+  #       key: SOME_CONFIGMAP_KEY
+
+
+# -- Allows to set additional environment variables based on simplified key-value syntax.
```
Comment (sagikazarmark) [inline charts/sftpgo/values.yaml:71]:
```suggestion
# -- Allows setting additional environment variables using a simplified key-value syntax.
# For advanced use cases, refer to `envVars`.
```

File: charts/sftpgo/templates/deployment.yaml [hunk 6]
```diff
@@ -38,6 +38,9 @@ spec:
             - sftpgo
             - serve
           env:
+            {{- with .Values.envVars }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
```
Comment (sagikazarmark) [inline charts/sftpgo/templates/deployment.yaml:43]: Leaving it above `env` is fine, but should be moved below the manually defined variables.

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Thanks @ggramal !

There is a few more things left, but looks good to me overall.

File: charts/sftpgo/templates/deployment.yaml (hunk 6 above)
Comment (ggramal) [reply charts/sftpgo/templates/deployment.yaml]: makes sense. Please check @sagikazarmark

File: charts/sftpgo/values.yaml (hunk 4 above)
Comment (ggramal) [reply charts/sftpgo/values.yaml]: done

File: charts/sftpgo/values.yaml (hunk 5 above)
Comment (ggramal) [reply charts/sftpgo/values.yaml]: done

**Review by sagikazarmark — APPROVED** [review]
LGTM, thanks @ggramal !

**Pull request merged**